Entities/Items/Projectiles/Arrow.as

```diff
@@ -165,22 +163,40 @@ void onTick(CBlob@ this)
 
 	// sticking
 	if (processSticking)
-	{
-		//no collision
-		shape.getConsts().collidable = false;
-
-		if (!this.hasTag("_collisions"))
-		{
-			this.Tag("_collisions");
-			// make everyone recheck their collisions with me
-			const uint count = this.getTouchingCount();
-			for (uint step = 0; step < count; ++step)
+	{	
+		if (this.get_bool("hitBlobExists")) 
+		{
+			// structure blob: door, platform, team bridge, trap block etc.
+			
+			CBlob@ gottenBlob = getBlobByNetworkID(this.get_u32("hitBlob"));
+			
+			if (gottenBlob is null) // structure blob is gone
 			{
-				CBlob@ _blob = this.getTouchingByIndex(step);
-				_blob.getShape().checkCollisionsAgain = true;
+				this.server_Die();
+			}
+			else if (gottenBlob.hasTag("fallen")										// structure blob is falling
+					|| (gottenBlob.getName().find("door") != -1 && isOpen(gottenBlob))	// opening door
+					|| (gottenBlob.getName() == "bridge" && isOpen(gottenBlob))			// opening bridge
+					|| (gottenBlob.getName() == "trap_block" && isOpen(gottenBlob)))	// opening trap block
+			{
```

Comment:
Checking if gottenBlob is open three times is redundant, it can be done with just one check. You will have to add another set of parenthesis that group your getName checks. Then just add the isOpen check at the end. Should be something like ``if (b.hasTag("tag") || ((namecheck1 || namecheck2 || namecheck3) && isOpen(b)))``

Reply:
Thank you. All of your suggestions have been addressed in [8f1846a](https://github.com/transhumandesign/kag-base/pull/1279/commits/8f1846a6a8c8d30cfbc241c86f9026684e938d0b).